### PR TITLE
[RAPTOR-3319] allow to enforce run language

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Allow to enforce run language with argument: --language [java, r, python]
This argument will be added into start_server.sh in the drop in environments

Added header "Running environment language: [R|Java|Python]" into error messages. Because of current code structure, I didn't find a nice way to add it into one common place.

## Rationale
This argument will be used in the drop in envs, to give env some control and ensure that drum will running proper framework, e.g. running Java in java env.

### CODE REVIEW

**Comment by yakov-g** [comment]
jarvis test functional_drop_in

File: custom_model_runner/datarobot_drum/drum/args_parser.py [hunk 1]
```diff
@@ -314,14 +314,20 @@ def _reg_arg_url(*parsers):
 
     @staticmethod
     def _reg_arg_language(*parsers):
-        langs = [e.value for e in RunLanguage]
-        langs.remove(RunLanguage.JAVA.value)
         for parser in parsers:
+            langs = [e.value for e in RunLanguage]
+            prog_name_lst = CMRunnerArgsRegistry._tokenize_parser_prog(parser)
```
Comment (klichukb) [inline custom_model_runner/datarobot_drum/drum/args_parser.py:319]: can you explain briefly what does this logic below means?

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/args_parser.py:319]:
```
@staticmethod
    def _tokenize_parser_prog(parser):
        # example:
        # - for score_parser prog is "drum score"
        # - for new_model_parser prog is "drum new model"
        return parser.prog.split(" ")
```
So i split prog by tokens to be able to customize arg's behavior for different subcommands.

Comment (klichukb) [reply custom_model_runner/datarobot_drum/drum/args_parser.py:319]: I see, but what about "if its a new model, remove java from langs"?

Comment (klichukb) [reply custom_model_runner/datarobot_drum/drum/args_parser.py:319]: so from DMs FYI, this means that this logic doesnt support creating a template for java for now.
As well we figured it out that we might start doing so for mojo/pojo.

File: tests/drum/test_custom_model.py [hunk 2]
```diff
@@ -476,6 +480,78 @@ def test_detect_language(self, framework, problem, language, tmp_path):
...
+            ),  # java and sklearn artifacts, no custom.py
+            (
+                CODEGEN_AND_SKLEARN,
+                REGRESSION,
+                NO_CUSTOM,
+                "r",
+            ),  # java and sklearn artifacts, no custom.py
+        ],
+    )
+    def test_set_language(self, framework, problem, language, set_language, tmp_path):
```
Comment (klichukb) [inline tests/drum/test_custom_model.py:520]: Very nice test

File: custom_model_runner/datarobot_drum/drum/args_parser.py (hunk 1 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/args_parser.py:319]: https://datarobot.atlassian.net/browse/RAPTOR-3330

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**4.45** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**